airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/paginators/strategies/offset_increment.py

```diff
@@ -60,7 +60,7 @@ def next_page_token(self, response: requests.Response, last_page_size: int, last
         decoded_response = self.decoder.decode(response)
 
         # Stop paginating when there are fewer records than the page size or the current page has no records
-        if (self._page_size and last_page_size < self._page_size.eval(self.config, response=decoded_response)) or last_page_size == 0:
+        if (self._page_size and last_page_size < self._page_size.eval(self.config, response=next(decoded_response))) or last_page_size == 0:
```

Comment:
is there a chance decoder.decode will return an empty iterable? If yes, let's catch the exception and raise a user-friendly error message